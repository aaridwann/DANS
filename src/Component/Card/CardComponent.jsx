import React from "react";

function CardComponent({ company, title, img, desc, type, onClick }) {
  return (
      <div className="pt-0 w-full items-center justify-center flex pr-4 pb-0 pl-4 mt-0 mr-auto mb-0 ml-auto max-w-7xl sm:px-6 lg:px-8">
        <div className="pt-0 pr-4 pb-0 pl-4 mt-0 mr-auto mb-0 ml-auto max-w-4xl sm:px-6 lg:px-8">
          <div className="shadow-xl w-[300px] md:w-[800px] bg-white mt-8 mr-0 mb-0 ml-0 pt-4 pr-10 pb-4 pl-10 flow-root rounded-lg sm:py-2">
            <div className="pt--10 pr-0 pb-10 pl-0">
              <div className="pt-5 pr-0 pb-0 pl-0 mt-5 mr-0 mb-0 ml-0">
                <div className="sm:flex sm:items-center sm:justify-between sm:space-x-5">
                  <div className="flex items-center flex-1 min-w-0">
                    <img
                      src="https://d34u8crftukxnk.cloudfront.net/slackpress/prod/sites/6/SlackLogo_CompanyNews_SecondaryAubergine_Hero.jpg?d=500x500&amp;f=fill"
                      className="flex-shrink-0 object-cover rounded-full btn- w-10 h-10"
                    />
                    <div className="mt-0 mr-0 mb-0 ml-4 flex-1 min-w-0">
                      <p onClick={onClick} className="text-lg cursor-pointer font-bold text-gray-800 truncate">
                        {/* Engineering Manager */}
                        {title}
                      </p>
                      <div className=" flex items-center justify-start gap-4">

                      <p className="text-gray-600 text-md">{company}</p>
                      <p className="text-gray-600 text-md">{type}</p>
                      </div>
                    </div>
                  </div>
                  <div className="mt-4 mr-0 mb-0 ml-0 pt-0 pr-0 pb-0 pl-14 flex items-center sm:space-x-6 sm:pl-0 sm:mt-0">
                    <a
                      href=""
                      className="bg-gray-800 pt-2 pr-6 pb-2 pl-6 text-lg font-medium text-gray-100 transition-all
                      duration-200 hover:bg-gray-700 rounded-lg"
                    >
                      Apply
                    </a>
                  </div>
                </div>
              </div>

            </div>
          </div>
        </div>
      </div>
  );
}

export default CardComponent;
