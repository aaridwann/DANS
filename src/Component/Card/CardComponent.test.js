import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CardComponent from "./CardComponent";
import "@testing-library/jest-dom/extend-expect";

describe("Card component testing unit", () => {
  test("Snapshot test", () => {
    expect(render(<CardComponent />)).toMatchSnapshot();
  });
  const props = {
    company: "Test Company",
    title: "Test Title",
    img: "https://test.com/test-img.jpg",
    desc: "Test Description",
    type: "Test Type",
    onClick: jest.fn(),
  };

  it("renders correctly with props", () => {
    const { getByText, getByAltText } = render(<CardComponent {...props} />);
    expect(getByText(props.company)).toBeInTheDocument();
    expect(getByText(props.title)).toBeInTheDocument();
    expect(getByText(props.type)).toBeInTheDocument();
    expect(getByText(props.title)).toBeInTheDocument();
  });

  it("calls onClick when title is clicked", () => {
    const { getByText } = render(<CardComponent {...props} />);
    fireEvent.click(getByText(props.title));
    expect(props.onClick).toHaveBeenCalled();
  });
});
