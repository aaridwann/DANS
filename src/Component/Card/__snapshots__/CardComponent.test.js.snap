// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component testing unit Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pt-0 w-full items-center justify-center flex pr-4 pb-0 pl-4 mt-0 mr-auto mb-0 ml-auto max-w-7xl sm:px-6 lg:px-8"
      >
        <div
          class="pt-0 pr-4 pb-0 pl-4 mt-0 mr-auto mb-0 ml-auto max-w-4xl sm:px-6 lg:px-8"
        >
          <div
            class="shadow-xl w-[300px] md:w-[800px] bg-white mt-8 mr-0 mb-0 ml-0 pt-4 pr-10 pb-4 pl-10 flow-root rounded-lg sm:py-2"
          >
            <div
              class="pt--10 pr-0 pb-10 pl-0"
            >
              <div
                class="pt-5 pr-0 pb-0 pl-0 mt-5 mr-0 mb-0 ml-0"
              >
                <div
                  class="sm:flex sm:items-center sm:justify-between sm:space-x-5"
                >
                  <div
                    class="flex items-center flex-1 min-w-0"
                  >
                    <img
                      class="flex-shrink-0 object-cover rounded-full btn- w-10 h-10"
                      src="https://d34u8crftukxnk.cloudfront.net/slackpress/prod/sites/6/SlackLogo_CompanyNews_SecondaryAubergine_Hero.jpg?d=500x500&f=fill"
                    />
                    <div
                      class="mt-0 mr-0 mb-0 ml-4 flex-1 min-w-0"
                    >
                      <p
                        class="text-lg cursor-pointer font-bold text-gray-800 truncate"
                      />
                      <div
                        class=" flex items-center justify-start gap-4"
                      >
                        <p
                          class="text-gray-600 text-md"
                        />
                        <p
                          class="text-gray-600 text-md"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="mt-4 mr-0 mb-0 ml-0 pt-0 pr-0 pb-0 pl-14 flex items-center sm:space-x-6 sm:pl-0 sm:mt-0"
                  >
                    <a
                      class="bg-gray-800 pt-2 pr-6 pb-2 pl-6 text-lg font-medium text-gray-100 transition-all duration-200 hover:bg-gray-700 rounded-lg"
                      href=""
                    >
                      Apply
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pt-0 w-full items-center justify-center flex pr-4 pb-0 pl-4 mt-0 mr-auto mb-0 ml-auto max-w-7xl sm:px-6 lg:px-8"
    >
      <div
        class="pt-0 pr-4 pb-0 pl-4 mt-0 mr-auto mb-0 ml-auto max-w-4xl sm:px-6 lg:px-8"
      >
        <div
          class="shadow-xl w-[300px] md:w-[800px] bg-white mt-8 mr-0 mb-0 ml-0 pt-4 pr-10 pb-4 pl-10 flow-root rounded-lg sm:py-2"
        >
          <div
            class="pt--10 pr-0 pb-10 pl-0"
          >
            <div
              class="pt-5 pr-0 pb-0 pl-0 mt-5 mr-0 mb-0 ml-0"
            >
              <div
                class="sm:flex sm:items-center sm:justify-between sm:space-x-5"
              >
                <div
                  class="flex items-center flex-1 min-w-0"
                >
                  <img
                    class="flex-shrink-0 object-cover rounded-full btn- w-10 h-10"
                    src="https://d34u8crftukxnk.cloudfront.net/slackpress/prod/sites/6/SlackLogo_CompanyNews_SecondaryAubergine_Hero.jpg?d=500x500&f=fill"
                  />
                  <div
                    class="mt-0 mr-0 mb-0 ml-4 flex-1 min-w-0"
                  >
                    <p
                      class="text-lg cursor-pointer font-bold text-gray-800 truncate"
                    />
                    <div
                      class=" flex items-center justify-start gap-4"
                    >
                      <p
                        class="text-gray-600 text-md"
                      />
                      <p
                        class="text-gray-600 text-md"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="mt-4 mr-0 mb-0 ml-0 pt-0 pr-0 pb-0 pl-14 flex items-center sm:space-x-6 sm:pl-0 sm:mt-0"
                >
                  <a
                    class="bg-gray-800 pt-2 pr-6 pb-2 pl-6 text-lg font-medium text-gray-100 transition-all duration-200 hover:bg-gray-700 rounded-lg"
                    href=""
                  >
                    Apply
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
